src/porepy/numerics/fv/upwind.py

```diff
@@ -226,7 +226,13 @@ def discretize(self, sd: pp.Grid, data: dict) -> None:
 
         # Get the sign of the advective flux.
         darcy_flux: np.ndarray = np.sign(parameter_dictionary[self._flux_array_key])
-        bc: pp.BoundaryCondition = parameter_dictionary["bc"]
+
+        # Enables the creation of an upwind object even if boundary data is not
+        # externally provided.
+        if "bc" in parameter_dictionary:
+            bc: pp.BoundaryCondition = parameter_dictionary["bc"]
+        else:
+            bc = pp.BoundaryCondition(sd, sd.get_boundary_faces(), "dir")
```

Comment:
Do you have specific reasons to set Dirichlet conditions by default? To me the choice is somewhat arbitrary, though I can note that in the `BoundaryCondition` class, Neumann conditions are enforced by default.

Reply:
I trust your judgement there and add your explanation as a comment in the code.